lib/sqlite.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const path = require('path')
+const { open } = require('sqlite')
+const sqlite3 = require('sqlite3')
+if (false) sqlite3.verbose()
+
+const mode = readwrite =>
+  readwrite ?
+    sqlite3.OPEN_READWRITE | sqlite3.OPEN_CREATE :
+    sqlite3.OPEN_READONLY // TODO: check if opening fails
```

Comment:
I forgot about this one... Just confirmed that opening in readonly mode fails if the file doesn't exist. This will be a bit of a pain to work around...